src/Components/Web.JS/src/Services/NavigationEnhancement.ts

```diff
@@ -116,13 +119,26 @@ function onDocumentSubmit(event: SubmitEvent) {
       return;
     }
 
+    const submitter = event.submitter as HTMLButtonElement;
+
+    if ((submitter && submitter.formMethod && submitter.formMethod === 'dialog') ||
+      formElem.method === 'dialog') {
```

Comment:
```suggestion
    if (submitter?.formMethod === 'dialog' || formElem.method === 'dialog') {
```
You might have written it this way for better browser compatibility, but I believe the JS produced by this will desugar into effectively what you've written here. Even if it doesn't, I think we should be okay to use `?.` because every major browser has supported it [since 2020](https://caniuse.com/mdn-javascript_operators_optional_chaining).